api/app/signals/apps/questionnaires/services/question_graph.py

```diff
@@ -45,12 +45,16 @@ def _build_nx_graph(q_graph, edges):
         # To allow for matching rule and default rule (i.e. a double edge).
         nx_graph = networkx.MultiDiGraph()
 
+        for _, question in q_graph.get_nodes().items():
+            nx_graph.add_node(question.id, label=question.label)
```

Comment:
As discussed, this loop can be removed. Adding all required metadata to the nodes can be achieved in the loop below if we do a `select_related` on `question` and `next_question` for the edges parameter passed into `_build_nx_graph`.

see patch for details